File: 프로그래머스/src/프로그래머스_02262022/소수찾기.cpp
#include <string>
#include <vector>

using namespace std;

//미리 처음에 구해놓으까?
vector<bool> isPrime(10000000, true);
int solution(int n)
{
    // false : 소수, true : 소수가 아님
    for (int i = 2; i * i < 1000000; i++)
    {
        if (isPrime[i]) //소수인 수 발견 -> 이제 이 숫자의
            for (int j = i * i; j <= 1000000; j += i)
            {
                isPrime[j] = false;
            }
    }

    int cnt = 0;

    for (int i = 2; i <= n; i++)
    {
        if (isPrime[i])
            cnt++;
    }
    return cnt;
}

File: 2026/완전탐색/소수찾기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/42839?language=python3

from itertools import *
def is_prime(num):
    if num == 0 or num == 1 :
        return False
    for x in range (2,num):
        if num % x == 0 :
            return False;
    return True;

def solution(numbers):
    answer = 0
    # set : without duplication
    vec = set() # this is how to declere set
    
    for times in range(len(numbers)):
        for x in permutations(numbers, times+1):
            # vec.set(''.join)
            if x[0] == '0':
                continue
            temp = ''.join(x)
            # print(temp)
            vec.add(temp) if is_prime(int(temp)) else None
    # print(vec)
    return len(vec)